### PR TITLE
Codelens

### PR DESCRIPTION
**Pull request by amirbilu**

### CODE REVIEW

**Comment by amirbilu** [comment]
This PR adds Tabnine code lens, a hint next to method / function signature which suggests that Tabnine can explain / fix / test the code. @aarondill can you please take a look? would appreciate also your feedback on the UX - as its not final.

https://github.com/codota/tabnine-nvim/assets/7681048/7d8dd17b-37d4-4db1-accc-0edd5af9152c

File: lua/tabnine/workspace.lua [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+	timer:start(
+		0,
+		30000,
+		vim.schedule_wrap(function()
+			tabnine_binary:request({ Workspace = { root_paths = { fn.getcwd() } } }, function() end)
+		end)
+	)
+end
+
+return M
```
Comment (amirbilu) [inline lua/tabnine/workspace.lua:18]: Not really related to this PR

**Comment by aarondill** [comment]
can we add a config toggle for this please? just something like `codelens_enabled` which defaults to true.
the other option is to add a `chat_enabled`, which also defaults to true, that codelens respects (since it's technically a Chat feature)


also, have you tested what happens when you use this *without* first compiling Chat?

**Comment by aarondill** [comment]
> @aarondill what do you generally feel about this feature? Would you use it?

I personally would not use it, however I don't use Chat either. I can see the appeal, but I use Tabnine solely as code completion/generation, instead of a general LLM.

**Pull request merged**